packages/kit/src/core/postbuild/fork.js

```diff
@@ -0,0 +1,72 @@
+import { fileURLToPath } from 'node:url';
+import child_process from 'node:child_process';
+
+/**
+ * @template T
+ * @template U
+ * @param {string} module
+ * @param {(opts: T) => U} callback
+ * @returns {(opts: T) => Promise<U>}
+ */
+export function forked(module, callback) {
+	if (process.env.SVELTEKIT_FORK) {
+		process.on(
+			'message',
+			/** @param {any} data */ async (data) => {
+				if (data.type === 'args' && data.module === module) {
```

Comment:
Do we need to be more careful here? If the user is also sending messages, I don't think we can assume that `data` will be an object, can we?

```suggestion
				if (data && data.type === 'args' && data.module === module) {
```